ra2ce/common/configuration/config_wrapper_protocol.py

```diff
@@ -23,17 +23,23 @@
 from __future__ import annotations
 
 from pathlib import Path
-from typing import Any, List, Optional, Protocol, runtime_checkable
+from typing import Any, Optional, Protocol, runtime_checkable
 
 from ra2ce.common.configuration.config_data_protocol import ConfigDataProtocol
 
 
 @runtime_checkable
 class ConfigWrapperProtocol(Protocol):  # pragma: no cover
     ini_file: Path
-    root_dir: Path
     config_data: Optional[ConfigDataProtocol] = None
-    graphs: List[Any] = []
+    graphs: Optional[dict] = None
+
+    @property
+    def root_dir(self) -> Path:
```

Comment:
this is already implemented?

Reply:
Should be implemented in the implementing classes. I made it a property to resolve issues at the implementing classes where it already was a (read-only) property.

Reply:
Proper docstring added.